main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <random>
+#include <regex>
+
+void roll(int numberDice, int diceSides);
+void parseString(char *argument);
```

Comment:
These should both be marked static (along with the actual function definitions below). Right now, these functions have global scope (and they don't need to).